### PR TITLE
get attandance user by Id & and show data in form

### PR DESCRIPTION
**Pull request by ghulamhafeez**
### Implementation

* Get attendance by userId and fill her data into form

### Resources

https://github.com/user-attachments/assets/3278fe2f-0312-4754-aea3-70f54bf9620e

### CODE REVIEW

**Review by muhammad-bilal-ali-khokhar — COMMENTED** [review]
@ghulamhafeez  pleaes impove PR Description

File: src/app/attendance/actions/getAttendanceById.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+export const getAttendanceById = async (id: number) => {
+  const supabase = await getSupabaseClient();
+
+  const { data } = await supabase
+    .from(Tables.Attendance)
+    .select("*")
+    .eq("id", id)
+    .single();
+
+  return {
```
Comment (muhammad-bilal-ali-khokhar) [inline src/app/attendance/actions/getAttendanceById.ts:15]: @ghulamhafeez must be error handling

File: src/lib/routes.tsx [hunk 2]
```diff
@@ -3,6 +3,8 @@ export enum Routes {
   Dashboard = "/dashboard",
   Notification = "/notification",
   Attendance = "/attendance",
+  AddAttendance = "/attendance/add",
+  EditAttendance = "/attendance/edit?id=id",
```
Comment (nadirhussain786) [inline src/lib/routes.tsx:7]: I think we should use `useParams` instead of relying on param queries for routes.

**Review by muhammad-bilal-ali-khokhar — APPROVED** [review]

File: src/app/attendance/actions/getAttendanceById.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+import { getSupabaseClient } from "@/utils/supabase/supabaseClient";
+
+export const getAttendanceById = async (id: number) => {
+  const supabase = await getSupabaseClient();
+
+  const { data } = await supabase
+    .from(Tables.Attendance)
+    .select("*")
+    .eq("id", id)
+    .single();
```
[the remark this answers is not in the document]
Comment (ghulamhafeez) [reply src/app/attendance/actions/getAttendanceById.ts:13]: While it’s not strictly necessary, it is helpful whenever You expect only one result (e.g., querying by a unique id).

File: src/lib/routes.tsx (hunk 2 above)
Comment (ghulamhafeez) [reply src/lib/routes.tsx]: Fixed

File: src/app/attendance/actions/getAttendanceById.ts (hunk 1 above)
Comment (ghulamhafeez) [reply src/app/attendance/actions/getAttendanceById.ts:15]: Fixed

**Pull request merged**